src/pykiso/lib/auxiliaries/zephyr.py

```diff
@@ -0,0 +1,223 @@
+import subprocess
+import xml.etree.ElementTree as ET
+from pathlib import Path
+from typing import Optional
+from pykiso import AuxiliaryInterface
+from pykiso.types import MsgType
+
+log = logging.getLogger(__name__)
+
+
+class TestResult(enum.IntEnum):
+    """Result of a twister test"""
+
+    PASSED = 0
+    FAILED = 1
+    SKIPPED = 2
+    ERROR = 3
+
+
+class ZyphyrException(Exception):
+    """Exception for twister errors"""
+
+    def __init__(self, message) -> None:
+        super().__init__(message)
+
+
+class Twister:
+    """Control class for the Zephyr twister commandline tool"""
+
+    def __init__(self, twister_path: str = "twister") -> None:
+        self.twister_path = twister_path
+        self.process: Optional[subprocess.Popen[bytes]] = None
+
+    def start_test(
+        self, test_directory: str, testcase_name: str, wait_for_start: bool = True
+    ) -> None:
+        """Start the test case
+
+        :param test_directory: The directory to search for the Zephyr test project (passed to twister using the -T parameter)
+        :param testcase_name: The name of the Zephyr test (passed to twister using the -s parameter)
+        :param wait_for_start: Wait for the test case to start on the target. If this is set to false the function will just return even if the test is not running yet.
+
+        :return: True if successful
+        """
+        if self.process != None:
+            raise ZyphyrException("Twister test is already running")
+
+        self.outdir = str(Path(test_directory).resolve() / "twister-out")
+        logging.info(f"Using Twister output directory: {self.outdir}")
+
+        self.process = subprocess.Popen(
+            [
+                self.twister_path,
+                "-vv",
+                "-T",
+                test_directory,
+                "-s",
+                testcase_name,
+                "--outdir",
+                self.outdir,
+                "--report-dir",
+                self.outdir,
+            ],
+            stderr=subprocess.PIPE,
+            shell=False,
+        )
+
+        if wait_for_start:
+            # Read twister output until test case was started
+            while True:
+                line = self._readline()
+                if line == None:
+                    return
+
+                if "OUTPUT: START - " in line:
+                    log.info(f"Zephyr test started.")
+                    return
+
+    def _readline(self) -> Optional[str]:
+        """Read a line from the twister process
+
+        :return: The line as string or None if the process has finished
+        """
+        while True:
+            read = self.process.stderr.readline()
+            if len(read) == 0:
+                return None
+            line = read.decode("utf-8").strip()
+            # Twister outputs many lines without a real message, remove those
+            output = line.find("- OUTPUT:")
+            if output != -1:
+                message = line[output + 9 :]
```

Comment:
To avoid the magic numbers:
```
test_output_pattern = "- OUTPUT"
output = line.find(test_output_pattern)
message = line[output + len(test_output_pattern) :]
```

Reply:
Done